runtime/src/bank.rs

```diff
@@ -3423,6 +3429,15 @@ impl Bank {
         &self,
         transaction: SanitizedTransaction,
     ) -> TransactionSimulationResult {
+        // Simulation detection countermeasure 2: Randomize the incinerator balance to make it more
+        // difficult for a program to detect if it's running in simulation.
+        //
+        // Reference: https://opcodes.fr/en/publications/2022-01/detecting-transaction-simulation/
+        let _ = self.deposit(
```

Comment:
I'm not convinced this is the right thing to do.  More often than not the incinerator balance is zero, the transaction history for the incinerator is mostly just confused sheeple thinking that sending an NFT to the incinerator actually does something: https://explorer.solana.com/address/1nc1nerator11111111111111111111111111111111
So if we randomize the incinerator balance during simulation, it seems like a check for a balance of zero would actually now create a pretty strong signal that the program is in fact running within a simulation!

Reply:
(off-topic sorry –) Speaking out as a member of the NFT incinerator sheeple. We do this because we're lazy and sending ancillary spam/scam NFTs there is so much easier UX-wise than creating a new derived token account, transferring it into the new account, and closing said account. Unless I'm missing some easier way of doing this in Solflare.

Reply:
> (off-topic sorry –) Speaking out as a member of the NFT incinerator sheeple. We do this because we're lazy and sending ancillary spam/scam NFTs there is so much easier UX-wise than creating a new derived token account, transferring it into the new account, and closing said account. Unless I'm missing some easier way of doing this in Solflare.

Please request that Solflare add support for the burn instruction.  This is The Way :tm: to be rid of unwanted SPL Tokens.  Sending them to some other random address is actually worse than the incinerator

Reply:
> I'm not convinced this is the right thing to do. More often than not the incinerator balance is zero, the transaction history for the incinerator is mostly just confused sheeple thinking that sending an NFT to the incinerator actually does something: https://explorer.solana.com/address/1nc1nerator11111111111111111111111111111111 So if we randomize the incinerator balance during simulation, it seems like a check for a balance of zero would actually now create a pretty strong signal that the program is in fact running within a simulation!

I suppose we'd have to get gross and randomize the actual incinerator balance on bank creation, stash that value and subtract it back out on freeze to do much better?

Reply:
I was thinking for max grossness we could shuttle a random % of the burned tx fees to the incinerator on each block.  But yeah, now we're off the rails and ultimately the wallets themselves need to protect users more by appending check instructions to transactions IMO